### PR TITLE
change masks to support head_importance

### PR DESCRIPTION
**Pull request by vasilisvyth**
I return also a list with the head_importance scores. Initially, I thought of returning predictions and labels as well but I don't think there is any value in it. You can freely change anything you don't like regarding the save location.

### CODE REVIEW

File: head_importance_analysis.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        mask = torch.isinf(tmp_importance_scores)
+
+    # Use the mask to extract non-infinity elements
+        non_infinity_elements = tmp_importance_scores[~mask]   
+        mean = torch.mean(non_infinity_elements).item()
+        std = torch.std(non_infinity_elements).item()
+        mean = round(mean,3)
+        std = round(std,3)
+        statistics.append(f'{mean}\u00B1{std}')
+#todo add plots for same language, same task, same seed and see how the head importance scores change with time
```
Comment (gergopool) [inline head_importance_analysis.py:28]: Is this file finished?
The for loop is not necessary, these can be done in vector operations.
Also, it would be nice to add argparse, similar to other runner scripts.

Comment (vasilisvyth) [reply head_importance_analysis.py:28]: leave this file just add it when you merge

File: head_importance_analysis.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+seed=0
+path = os.path.join('./results/importance_scores', task, f'head_imp_en_3.pickle')
+#inf value when head_importance = 0 i.e. it is already removes
+with open(path,'rb') as file:
+    importance_scores = pickle.load(file)
+    #! they are in [0,1] interval so they are a bit useless these statistics
+    statistics = [] 
+    for i in range(len(importance_scores)):
+        tmp_importance_scores = importance_scores[i].flatten()
+        mask = torch.isinf(tmp_importance_scores)
```
[the remark this answers is not in the document]
Comment (vasilisvyth) [reply head_importance_analysis.py:19]: The author in his code puts infinite values to what he has already selected so it's like having 0 value. The author had put [float('inf')](https://github.com/apostolikas/Language-Specific-Subnetworks/blob/head_importance/bert_experiments/find_head_masks.py#L155) to distinguish the already pruned masks.
